### PR TITLE
Updating Chips

### LINKED ISSUE #1247: 2.0 - Update Chip
This is a tracking issue for updating Chip

**Scope:** Chip

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #1247 
Complies with #1293

- Added Chipset
- Fixed Exports
- Added `onSelectionChange` event.
- Added `proxyEventHandler` for ease of #1293

### CODE REVIEW

File: packages/base/src/MaterialComponent.ts [hunk 1]
```diff
@@ -126,6 +126,24 @@ export abstract class MaterialComponent<
     }
   }
 
+  protected proxyEventHandler(eventName: string, e: Event, options?: any) {
```
Comment (cromefire) [inline packages/base/src/MaterialComponent.ts]: Instead of `eventName: String` rather use `eventName: keyof MaterialComponentProps<PropType>`

Comment (prateekbh) [reply packages/base/src/MaterialComponent.ts]: TIL

Comment (prateekbh) [reply packages/base/src/MaterialComponent.ts]: done

File: packages/base/src/MaterialComponent.ts [hunk 2]
```diff
@@ -126,6 +126,24 @@ export abstract class MaterialComponent<
...
+    if (
+      eventName in this.props &&
+      typeof this.props[eventName] === 'function'
+    ) {
+      this.props[eventName](
+        e,
+        Object.assign(
+          {},
```
[the remark this answers is not in the document]
Comment (prateekbh) [reply packages/base/src/MaterialComponent.ts]: my bad

**Pull request merged**